homeassistant/components/kef/media_player.py

```diff
@@ -128,6 +139,81 @@
+            (d for d in hass.data[DOMAIN].values() if d.entity_id == entity_id), None,
+        )
+
+        if media_player is None:
+            _LOGGER.warning("Unable to find KEF speaker with entity_id: %s", entity_id)
+            return
+
+        if service.service == SERVICE_MODE:
+            await media_player.set_mode(
+                desk_mode=service.data.get("desk_mode"),
+                wall_mode=service.data.get("wall_mode"),
+                phase_correction=service.data.get("phase_correction"),
+                high_pass=service.data.get("high_pass"),
+                sub_polarity=service.data.get("sub_polarity"),
+                bass_extension=service.data.get("bass_extension"),
+            )
+        elif service.service == SERVICE_DESK_DB:
+            await media_player.set_desk_db(service.data.get("db"))
+        elif service.service == SERVICE_WALL_DB:
+            await media_player.set_wall_db(service.data.get("db"))
+        elif service.service == SERVICE_TREBLE_DB:
+            await media_player.set_treble_db(service.data.get("db"))
+        elif service.service == SERVICE_HIGH_HZ:
+            await media_player.set_high_hz(service.data.get("hz"))
+        elif service.service == SERVICE_LOW_HZ:
+            await media_player.set_low_hz(service.data.get("hz"))
+        elif service.service == SERVICE_SUB_DB:
+            await media_player.set_sub_db(service.data.get("db"))
+        elif service.service == SERVICE_UPDATE_DSP:
+            await media_player.update_dsp()
+
+    mode_schema = vol.Schema(
+        {
+            vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+            vol.Optional("desk_mode"): cv.boolean,
+            vol.Optional("wall_mode"): cv.boolean,
+            vol.Optional("phase_correction"): cv.boolean,
+            vol.Optional("high_pass"): cv.boolean,
+            vol.Optional("sub_polarity"): vol.In(["-", "+"]),
+            vol.Optional("bass_extension"): vol.In(["Less", "Standard", "Extra"]),
+        }
+    )
+    hass.services.async_register(
+        DOMAIN, SERVICE_MODE, service_handler, schema=mode_schema,
+    )
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_UPDATE_DSP,
+        service_handler,
+        schema=vol.Schema({vol.Required(ATTR_ENTITY_ID): cv.entity_id}),
+    )
+
+    def add_service(name, which, option):
+        schema = vol.Schema(
+            {
+                vol.Required(ATTR_ENTITY_ID): cv.entity_id,
+                vol.Required(option): vol.In(DSP_OPTION_MAPPING[which]),
+            }
+        )
+        hass.services.async_register(
+            DOMAIN, name, service_handler, schema=schema,
+        )
+
+    add_service(SERVICE_DESK_DB, "desk_db", "db")
+    add_service(SERVICE_WALL_DB, "wall_db", "db")
+    add_service(SERVICE_TREBLE_DB, "treble_db", "db")
+    add_service(SERVICE_HIGH_HZ, "high_hz", "hz")
+    add_service(SERVICE_LOW_HZ, "low_hz", "hz")
+    add_service(SERVICE_SUB_DB, "sub_db", "db")
```

Comment:
Use this approach https://github.com/home-assistant/core/blob/cd57b764ce9b0f14568ff1c3da8893450b832e70/homeassistant/components/sonos/media_player.py#L233-L241

Reply:
Should I use `platform = entity_platform.current_platform.get()` to get `platform` in this case?

Reply:
Yes

Reply:
Much nicer like that!